src/planner/expand_hypertable.c

```diff
@@ -423,55 +453,32 @@ transform_time_bucket_comparison(PlannerInfo *root, OpExpr *op)
 		switch (tce->type_id)
 		{
 			case INT2OID:
-				integralValue = const_datum_get_int(castNode(Const, value));
-				integralWidth = const_datum_get_int(width);
-
-				if (integralValue >= PG_INT16_MAX - integralWidth)
-					return op;
-
-				datum = Int16GetDatum(integralValue + integralWidth);
-				subst = (Expr *) makeConst(tce->type_id,
-										   -1,
-										   InvalidOid,
-										   tce->typlen,
-										   datum,
-										   false,
-										   tce->typbyval);
-				break;
-
 			case INT4OID:
-				integralValue = const_datum_get_int(castNode(Const, value));
-				integralWidth = const_datum_get_int(width);
-
-				if (integralValue >= PG_INT32_MAX - integralWidth)
-					return op;
-
-				datum = Int32GetDatum(integralValue + integralWidth);
-				subst = (Expr *) makeConst(tce->type_id,
-										   -1,
-										   InvalidOid,
-										   tce->typlen,
-										   datum,
-										   false,
-										   tce->typbyval);
-				break;
 			case INT8OID:
 				integralValue = const_datum_get_int(castNode(Const, value));
 				integralWidth = const_datum_get_int(width);
 
-				if (integralValue >= PG_INT64_MAX - integralWidth)
+				if (integralValue >= ts_time_get_max(tce->type_id) - integralWidth)
```

Comment:
Is this correct  in using max timestamp value inclusive?

Note that max is a valid timestamp and TS_TIMESTAMP_END is the value after TS_TIMESTAMP_MAX. TS_TIMESTAMP_NOEND is +infinity

Reply:
this is the int branch so TS_TIMESTAMP stuff does not apply

Reply:
Sorry, I accidentally put the comment in the wrong branch. My question is still valid though because it was meant to highlight a difference between the integer and timestamp branches. Since integer MAX is a valid value (as opposed to TS_TIMESTAMP_END used in the timestamp branch) and the check is inclusive, you will exit here for a valid time value while you will not in the timestamp branch where the inclusive check is using TS_TIMESTAMP_END, which is not a valid time value.